rancher/rancher
# Remove env on unset

## Issue: 

https://github.com/rancher/rancher/issues/37998

## Problem

Prior, when a setting was configured using an env var of the format
CATTLE_<setting-name>, the source was set to "env". That source
would then persist even when the env var was unset in the future.
This prevented users from ever changing a setting value through the
rancher API or UI.

To clarify: After configuring a setting's value with an env var, that value should persist even once the env var is unset. However, the user should be able to change it once it has been unset.

## Solution

Now, unsetting the env var during an upgrade
will cause the source to be reset. This will allow users to change
the value again as long as the setting is no longer longer set by
an env var at setup.
 
## Testing


## Engineering Testing
### Manual Testing

Configure setting CATTLE_EKS_REFRESH_CRON using env var. Tried to change value via UI. Was unable to which is expected. Then unset the env var. Should be able to update the env var with UI (should not need to use kubectl), was not able to. After fix I was able to change the env with UI after unsetting the env var or setting the value to empty string "".

### Automated Testing

Added unit tests to cover updating source to "env", unsetting source from "env", updating value when it should, not updating value when it shouldn't, updating default when changed.

## QA Testing Considerations

Setup rancher installation.
Restart/upgrade rancher using env var such as CATTLE_EKS_REFRESH_CRON to set a setting that is present in the setting UI.
Try to change the value in the UI, should not be possible. The option will not be available in UI but even the API should not work for this.
Restart/upgrade rancher unsetting the env var. The value of the setting should be the same but you should now be able to change it using the API and UI.

 
### Regressions Considerations
